lib/std/otel/Context.zig

```diff
@@ -0,0 +1,125 @@
+//! A type designed to propagate OpenTelemetry between different APIs in a loosely
+//! coupled way.
+//!
+//! Immutable.
+const Context = @This();
+
+/// The value used by the implementation to implement `Context`. No peeking!
+impl: std.otel.types.context.Context,
```

Comment:
[don't name things "context", don't name things "impl", don't have the same name twice](https://ziglang.org/documentation/master/#Avoid-Redundancy-in-Names)

Reply:
- I could move the `Context` and `AttachToken` up to a toplevel field of `otel.Types`, e.g. `otel.types.Context` and `otel.types.ContextAttachToken`
- `impl` could be renamed to `inner` or `concrete_type` (though those names seem just a vague TBH). Fundamentally this is a type meant to hide which implementation is being used, so that developers can target the API instead of a specific implementation.

I'm a bit stumped on this to be honest. [The specification refers to this type as `Context`](https://opentelemetry.io/docs/specs/otel/context/), and it is [specifically the OpenTelemetry `Context`](https://ziglang.org/documentation/master/#Avoid-Redundant-Names-in-Fully-Qualified-Namespaces). We can brainstorm a new name for the OpenTelemetry `Context`, but I'm not sure that would actually make things more clear.

Reply:
`OpenTelemetryContext`? `OpaqueSignalKeyValueStore`? `ActiveSignalValues`? `ActiveSignalBundle`?

Reply:
That's what I'm leaning towards too.

Reply:
I think you are struggling with the name because the API design is bad. You have things called "inner", "impl", and "Context" because there is too much object-oriented programming nonsense here. If you start deleting crap the names will start to be easier to come up with.

>A type designed to propagate OpenTelemetry between different APIs in a loosely coupled way.

??? just delete this, this doesn't sound useful at all.

Reply:
Below I try to explain why I think they have their `Context` type, but I am mostly interpreting your response as "don't follow the OpenTelemetry API specification". Which means I should spend more time reviewing APIs similar to Tracy and less time trying to polish this.

Feel free to close this PR.

---

Reviewing the OpenTelemetry specifications, the Context type boils down to this:

```zig
pub const Context = struct {
    active_span: ?otel.Span,
    baggage: std.StringHashMapUnmanaged([]const u8),
    // future telemetry signals go here
};
```

(Don't ask me why they created `baggage`, it seems like they were very worried about developers giving developers ways to pass around arbitrary data.)

Then when you create a child context you copy the parent context and update the value you changed:

```zig
const child_span = tracer.createSpan("child span", parent_context);
var child_context = parent_context;
child_context.active_span = child_span;
```

Context is separate from a callstack so they can model concurrency or threadpools or firing off a boatload of microservice requests I guess. And parent `Context` is separate from parent `Span` so implementations can pull together different types of telemetry.

Also keep in mind that OpenTelemetry implementations are supposed to implement stuff like setting http headers so  that context can be retained even across network requests.

I think the [OpenTelemetry profiles data model](https://github.com/open-telemetry/oteps/blob/main/text/profiles/0239-profiles-data-model.md) is closer to what Tracy does, but as I understand it OpenTelemetry is more interested in developing a profiling data storage format than an API.